n3fit/src/n3fit/tests/test_hyperopt.py

```diff
@@ -1,13 +1,87 @@
 """
     Test hyperoptimization features
 """
+import json
+import pathlib
+import shutil
+import subprocess as sp
 
 from numpy.testing import assert_approx_equal
+
 from n3fit.hyper_optimization import rewards
 
+
 def test_rewards():
-    """ Ensure that rewards continue doing what they are supposed to do """
+    """Ensure that rewards continue doing what they are supposed to do"""
     losses = [0.0, 1.0, 2.0]
     assert_approx_equal(rewards.average(losses), 1.0)
     assert_approx_equal(rewards.best_worst(losses), 2.0)
     assert_approx_equal(rewards.std(losses), 0.816496580927726)
+
+
+REGRESSION_FOLDER = pathlib.Path(__file__).with_name("regressions")
+QUICKNAME = "quickcard"
+EXE = "n3fit"
+REPLICA = "1"
+
+
+def load_data(info_file):
+    """Loads the information of the fit from the json files"""
+    with open(info_file, "r", encoding='utf-8') as file:
+        return json.load(file)
+
+
+def test_restart_from_pickle(tmp_path):
+    """Ensure that our hyperopt restart works as expected"""
+    # Prepare the run
+    quickcard = f"hyper-{QUICKNAME}.yml"
+    quickpath = REGRESSION_FOLDER / quickcard
+    # Set up some options
+    n_trials_stop = 2
+    n_trials_total = 4
+    output_restart = (
+        REGRESSION_FOLDER / f"run_{n_trials_stop}_trials_and_then_{n_trials_total}_trials"
+    )
+    output_direct = REGRESSION_FOLDER / f"run_{n_trials_total}_trials"
+
+    # cp runcard to tmp folder
+    shutil.copy(quickpath, tmp_path)
+    # run some trials for the first time
+    sp.run(
+        f"{EXE} {quickpath} {REPLICA} --hyperopt {n_trials_stop} " f"-o {output_restart}".split(),
+        cwd=tmp_path,
+        check=True,
+    )
+    # restart and calculate more trials
+    sp.run(
+        f"{EXE} {quickpath} {REPLICA} --hyperopt {n_trials_total} "
+        f"-o {output_restart} --restart".split(),
+        cwd=tmp_path,
+        check=True,
+    )
+    # start again and calculate all trials at once
+    sp.run(
+        f"{EXE} {quickpath} {REPLICA} --hyperopt {n_trials_total} " f"-o {output_direct}".split(),
+        cwd=tmp_path,
+        check=True,
+    )
+
+    # read up generated json files
+    restart_json_path = f"{output_restart}/nnfit/replica_{REPLICA}/tries.json"
+    restart_json = load_data(restart_json_path)
+    direct_json_path = f"{output_direct}/nnfit/replica_{REPLICA}/tries.json"
+    direct_json = load_data(direct_json_path)
+
+    # Remove the 'output_restart' and 'output_direct' directories
+    shutil.rmtree(output_restart, ignore_errors=True)
+    shutil.rmtree(output_direct, ignore_errors=True)
```

Comment:
Why not create `output_restart` and `output_direct` as sub-directories under `tmp_path` and hence avoiding removing them manually?

Reply:
Done. Thanks!